### PR TITLE
feat: add editing links

### PR DESCRIPTION
**Pull request by nastatsan**

### CODE REVIEW

File: client/js/Utils/sc_helper.js [hunk 1]
```diff
@@ -144,3 +144,60 @@ ScHelper.prototype.setLinkFormat = async function (addr, format) {
...
+    new sc.ScAddr(keynode),
+    [sc.ScType.EdgeAccessVarPosPerm, "_edgeFormat"],
+    new sc.ScAddr(window.scKeynodes["nrel_format"]),
+);
+let result = await scClient.templateSearch(template);
+if (!result.length) return null;
+return result[0].get("_edgePng");
+};
+
+ScHelper.prototype.templateCheckSystIdtf = async (link) => {
```
Comment (smousy) [inline client/js/Utils/sc_helper.js:193]: Is this function using anywhere?

File: components/scg/src/scg.js [hunk 2]
```diff
@@ -654,17 +707,53 @@ SCg.Editor.prototype = {
...
+        //     );
+        //     const resSection = await window.scClient.templateSearch(sectionTemplate);
+
+        //     return res.length !== 0 || resSection.length !== 0;
+        // };
+
+        // const checkCanEdit = async (addr) => {
+        //     return !(await checkCanDelete(addr));
+        // }
+        // if (this.canEdit) {
```
Comment (smousy) [inline components/scg/src/scg.js:734]: please, delete comments

File: components/scg/src/scg.js [hunk 3]
```diff
@@ -675,9 +764,13 @@ SCg.Editor.prototype = {
...
                     this.showTool(this.toolSetContent());
                 }
+            } else if (this.scene.selected_objects.length === 1) {
+                if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
+                    this.showTool(this.toolSetContent());
+                }
             }
-            if (this.scene.selected_objects.length > 0) this.showTool(this.toolDelete());
-        }
+            // if (this.scene.selected_objects.length > 0) this.showTool(this.toolDelete());
```
Comment (smousy) [inline components/scg/src/scg.js:772]: please delete comments

**Review by smousy — CHANGES_REQUESTED** [review]

**Review by smousy — COMMENTED** [review]
please format changed files

File: components/scg/static/components/html/scg-tools-panel.html [hunk 4]
```diff
@@ -30,7 +30,7 @@
                 data-html="true" data-placement="right" data-toggle="popover">
             <img src="static/components/images/scg/tool-change-idtf.png"/>
         </label>
-        <label id="scg-tool-set-content" type='button' class="btn btn-default sc-no-default-cmd btn-xs hidden" sc_control_sys_idtf="ui_scg_control_tool_set_content"
+        <label id="scg-tool-set-content" type='button' class="btn btn-default sc-no-default-cmd btn-xs" sc_control_sys_idtf="ui_scg_control_tool_set_content"
```
[the remark this answers is not in the document]
Comment (nastatsan) [reply components/scg/static/components/html/scg-tools-panel.html:33]: yes

**Review by KovalM — CHANGES_REQUESTED** [review]

**Review by KovalM — APPROVED** [review]

**Review by smousy — APPROVED** [review]

**Review by AlexandrZagorskiy — COMMENTED** [review]
update changelog

File: components/scg/src/scg.js [hunk 5]
```diff
@@ -654,29 +707,31 @@ SCg.Editor.prototype = {
...
-                } else if (this.scene.selected_objects[0] instanceof SCg.ModelContour) {
-                    this.showTool(this.toolChangeIdtf());
-                } else if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
-                    this.showTool(this.toolSetContent());
-                }
+    onSelectionChanged: async function () {
+        this.hideTool(this.toolChangeIdtf());
+        this.hideTool(this.toolChangeType());
+        this.hideTool(this.toolSetContent());
+        this.hideTool(this.toolDelete());
```
Comment (MikhailSadovsky) [inline components/scg/src/scg.js:714]: Here you changed existing delete logic. Delete button will be never shown, is this expected behavior?

File: components/scg/src/scg.js [hunk 6]
```diff
@@ -654,27 +707,30 @@ SCg.Editor.prototype = {
...
+            } else if (this.scene.selected_objects[0] instanceof SCg.ModelContour) {
+                this.showTool(this.toolChangeIdtf());
+            } else if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
+                this.showTool(this.toolSetContent());
+            }
+        } else if (this.scene.selected_objects.length === 1) {
+            if (this.scene.selected_objects[0] instanceof SCg.ModelLink) {
+                this.showTool(this.toolSetContent());
             }
             if (this.scene.selected_objects.length > 0) this.showTool(this.toolDelete());
```
Comment (MikhailSadovsky) [inline components/scg/src/scg.js:735]: This condition should go on level up. We can select multiple objects with Ctrl and delete button should be available as exist

**Review by AlexandrZagorskiy — APPROVED** [review]
Don't forget to rebase, conflict in changelog

File: docs/changelog.md [hunk 7]
```diff
@@ -6,6 +6,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 ### Added
+- Add editing links with saving them
```
Comment (MikhailSadovsky) [inline docs/changelog.md:9]:
```suggestion
- Add ability to edit links and save changes
```

**Review by MikhailSadovsky — APPROVED** [review]

**Pull request merged**